cmd/gitannex/gitannex.go

```diff
@@ -0,0 +1,515 @@
+// Package gitannex provides the "git-annex" command, aka "garrgoyle". This
+// command enables [git-annex] to communicate with rclone.
+//
+// Garrgoyle is an implementation of git-annex's [external special remote
+// protocol].  This protocol is line delimited and spoken over stdin and stdout.
+//
+// # Milestones
+//
+// (Tracked in [issue #7625].)
+//
+//  1. ✅ Minimal support for the [external special remote protocol]. Tested on
+//     "local" and "drive" backends.
+//  2. Add support for the ASYNC protocol extension. This may improve performance.
+//  3. Support the [simple export interface]. This will enable `git-annex
+//     export` functionality.
+//  4. Once the draft is finalized, support import/export interface.
+//
+// [git-annex]: https://git-annex.branchable.com/
+// [external special remote protocol]: https://git-annex.branchable.com/design/external_special_remote_protocol/
+// [simple export interface]: https://git-annex.branchable.com/design/external_special_remote_protocol/export_and_import_appendix/
+// [issue #7625]: https://github.com/rclone/rclone/issues/7625
+package gitannex
+
+import (
+	"bufio"
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rclone/rclone/cmd"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/cache"
+	"github.com/rclone/rclone/fs/operations"
+	"github.com/spf13/cobra"
+)
+
+const commandName string = "gitannex"
+const garrgoyleName string = "git-annex-remote-rclone-goyle"
```

Comment:
`goyle` is cute but may be more boring `native` or `builtin` thus to signal it's intimate relationship to rclone itself?

Reply:
I have to say I don't understand what "goyle" or "garrgoyle" means at all! Do we need to expose it to users?

Reply:
So I was trying to be clever with the name, but I've gotten pretty consistent feedback that the name is confusing!

We need to expose *some* name to users since they'll need to install a symlink somewhere (unless/until @joeyh adds a special-case to git-annex!).

I'll rebrand it as git-annex-remote-rclone-goyle and get rid of the "garrgoyle" name. Thanks, folks!